src/Sulu/Bundle/SnippetBundle/EventListener/CacheInvalidationSubscriber.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SnippetBundle\EventListener;
+
+use Sulu\Bundle\HttpCacheBundle\Cache\CacheManager;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetModifiedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetRemovedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\WebspaceDefaultSnippetModifiedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\WebspaceDefaultSnippetRemovedEvent;
+use Sulu\Bundle\SnippetBundle\Snippet\DefaultSnippetManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class CacheInvalidationSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var CacheManager|null
+     */
+    private $cacheManager;
+
+    /**
+     * @var DefaultSnippetManagerInterface
+     */
+    private $defaultSnippetManager;
+
+    public function __construct(
+        ?CacheManager $cacheManager,
+        DefaultSnippetManagerInterface $defaultSnippetManager
+    ) {
+        $this->cacheManager = $cacheManager;
+        $this->defaultSnippetManager = $defaultSnippetManager;
+    }
```

Comment:
Maybe you could use constructor property promotion for this. Then the properties also have the right type and you don't need the doc-comments like so:
```suggestion
    public function __construct(
        private ?CacheManager $cacheManager,
        private DefaultSnippetManagerInterface $defaultSnippetManager
    ) {
    }
```

Reply:
Depending if we also want this in 2.4, then we would have to support also php 7.4 😅

Reply:
Yeah, you mean 7.2 because in 7.4 we would at least have typed properties. The php language has grown quite a bit in a few versions.